File: yami4/src/tools/yami4idl/examples/cpp/print/client_synchronous.cpp
#include "print.h"
#include <yami4-cpp/yami.h>

#include <cstdlib>
#include <iostream>

int main(int argc, char * argv[])
{
    if (argc != 2)
    {
        std::cout
            << "expecting one parameter: server destination\n";
        return EXIT_FAILURE;
    }

    const std::string server_address = argv[1];

    try
    {
        yami::agent client_agent;
        
        print::printer my_printer(
            client_agent, server_address, "printer");

        // read lines of text from standard input
        // and post each one for transmission

        std::string input_line;

        while (std::getline(std::cin, input_line))
        {
            print::text msg;
            
            msg.content = input_line;

            // message with confirmation      
            my_printer.print_synchronously(msg);
        }
    }
    catch (const std::exception & e)
    {
        std::cout << "error: " << e.what() << std::endl;
    }
}

File: yami4/src/tools/yami4idl/examples/java/print/ClientSynchronous.java
import com.inspirel.yami.Agent;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class ClientSynchronous {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println(
                "expecting one parameter: server destination");
            return;
        }

        String serverAddress = args[0];

        try {
            Agent clientAgent = new Agent();
            
            print.Printer myPrinter = new print.Printer(
                clientAgent, serverAddress, "printer");

            // read lines of text from standard input
            // and post each one for transmission

            BufferedReader reader = new BufferedReader(
                new InputStreamReader(System.in));
            String inputLine;
            while (
                (inputLine = reader.readLine()) != null) {

                print.Text msg = new print.Text();
                msg.content = inputLine;

                // the "content" field name is arbitrary,
                // but needs to be recognized at the server side

                myPrinter.printSynchronously(msg);
            }

            clientAgent.close();

        } catch (Exception ex) {
            System.out.println(
                "error: " + ex.getMessage());
        }
    }
}
